web/src/repo/settings/RepoSettingsCodeIntelligencePage.tsx

```diff
@@ -0,0 +1,125 @@
+import * as GQL from '../../../../shared/src/graphql/schema'
+import * as React from 'react'
+import { eventLogger } from '../../tracking/eventLogger'
+import { PageTitle } from '../../components/PageTitle'
+import { RouteComponentProps } from 'react-router'
+import { fetchLsifDumps } from './backend'
+import { FilteredConnection, FilteredConnectionQueryArgs } from '../../components/FilteredConnection'
+import { Timestamp } from '../../components/time/Timestamp'
+import { Link } from '../../../../shared/src/components/Link'
+import { catchError, map } from 'rxjs/operators'
+import { Observable } from 'rxjs'
+import { LoadingSpinner } from '@sourcegraph/react-loading-spinner'
+import { sortBy } from 'lodash'
+import { ErrorLike } from '@sourcegraph/codeintellify/lib/errors'
+import { isErrorLike } from '../../../../shared/src/util/errors'
+import { useObservable } from '../../util/useObservable'
+
+interface LsifDumpNodeProps {
+    node: GQL.ILSIFDump
+}
+
+class LsifDumpNode extends React.PureComponent<LsifDumpNodeProps, {}> {
+    public render(): JSX.Element | null {
+        return (
+            <div className="lsif-dump list-group-item">
+                <div className="lsif-dump__row lsif-dump__main">
+                    <div className="lsif-dump__meta">
+                        <div className="lsif-dump__meta-root">
+                            <Link to={this.props.node.projectRoot.url}>
+                                <strong>{this.props.node.projectRoot.path}</strong>
+                            </Link>{' '}
+                            <small className="text-muted lsif-dump__meta-commit">
+                                <code>{this.props.node.projectRoot.commit.abbreviatedOID}</code>
+                            </small>
+                        </div>
+                    </div>
+
+                    <small className="text-muted lsif-dump__meta-timestamp">
+                        <Timestamp noAbout={true} date={this.props.node.uploadedAt} />
+                    </small>
+                </div>
+            </div>
+        )
+    }
+}
+
+class FilteredLsifDumpsConnection extends FilteredConnection<{}, LsifDumpNodeProps> {}
+
+interface Props extends RouteComponentProps<any> {
+    repo: GQL.IRepository
+}
+
+/**
+ * The repository settings code intelligence page.
+ */
+export const RepoSettingsCodeIntelligencePage: React.FunctionComponent<Props> = ({ repo, ...props }) => {
+    React.useEffect(() => eventLogger.logViewEvent('RepoSettingsCodeIntelligence'), [])
+
+    const dumpsOrError = useObservable(
+        React.useMemo(
+            () =>
+                fetchLsifDumps({ repository: repo.id, isLatestForRepo: true, first: 5000 }).pipe(
+                    map(({ nodes }: { nodes: GQL.ILSIFDump[] }) => sortBy(nodes, node => node.projectRoot.path)),
+                    catchError((error: ErrorLike) => [error])
```

Comment:
In JS, you can throw non-Error things, so `ErrorLike` is not correct here for the parameter.
```suggestion
                    catchError((error): [ErrorLike] => [asError(error)])
```